recipe/bld.bat

```diff
@@ -24,3 +24,5 @@ copy prog\udunits2.exe %SCRIPTS%\udunits2.exe || exit 1
 copy lib\udunits2.dll %SCRIPTS%\udunits2.dll || exit 1
 copy lib\*.exp %LIBRARY_LIB%\ || exit 1
 copy lib\*.lib %LIBRARY_LIB%\ || exit 1
+
+dir %LIBRARY_PREFIX%\share\udunits\
```

Comment:
That looks to me like prefix replacement that is not happening.  Pretty suspicious here:

```
(C:\bld\udunits2_1496255512077\_t_env) (root) C:\bld\udunits2_1496255512077\test_tmp>udunits2 -h 
Usage:
    udunits2 -h
    udunits2 [-A|-L|-U] [-r] [-H have] [-W want] [XML_file]
where:
    -A         Use ASCII encoding (default).
    -L         Use ISO-8859-1 (ISO Latin-1) encoding.
    -U         Use UTF-8 encoding.
    -h         Help.  Print this message.
    -r         Reveal any problems in the database.
    -H have    Use "have" unit for conversion. Default is reply to prompt.
    -W want    Use "want" unit for conversion. Empty string requests
               definition of "have" unit. Default is reply to prompt.
    XML_file   XML database file. Default is "C:\bld\udunits2_1496255512077\_b_env\Library\share\udunits\udunits2.xml".
```

Looks to me like the bad prefix is hardcoded somehow into your executable.  That really sucks, because conda does not presently do binary replacement on Windows, other than Pip entry point exe's.  I recommend looking in the udunits2 source code to see if you can avoid embedding this path, or at least set it to a relative path.